src/Storages/StorageFile.cpp

```diff
@@ -565,6 +565,7 @@ class StorageFileSource : public ISource
         , block_for_format(block_for_format_)
         , context(context_)
         , max_block_size(max_block_size_)
+        , disable_parallel_parsing(storage->paths.size() > 1)
```

Comment:
We should not simply disable the parallel parsing but lower the number of threads accordingly.

See StorageMerge.cpp, `size_t current_streams = ...`

For example, if `max_threads = 16` and we have 8 files, every file will be parsed with 2 threads.

PS. @nikitamikhaylov is it possible?

Reply:
Yes, it is possible, need just to pass the number of threads to use in `getInputFormat` and then to `ParallelParsingInputFormat` function which looks a little bit sloppy or pass the number of "streams" and modify logic inside the function https://github.com/ClickHouse/ClickHouse/blob/2ae0b43570a50dbb0da7005e2c14374ef306969f/src/Formats/FormatFactory.cpp#L232-L275

Reply:
updated the function by passing number of streams parameters and calculating max threads